exchanges/bitfinex/bitfinex_test.go

```diff
@@ -992,40 +989,168 @@ func TestGetDepositAddress(t *testing.T) {
 		t.Error(err)
 	}
 	timer := time.NewTimer(sharedtestvalues.WebsocketResponseDefaultTimeout)
 	select {
 	case resp := <-b.Websocket.DataHandler:
-		if resp.(map[string]interface{})["event"] != "auth" && resp.(map[string]interface{})["status"] != "OK" {
-			t.Error("expected successful login")
+		if logResponse, ok := resp.(map[string]interface{}); ok {
+			if logResponse["event"] != "auth" && logResponse["status"] != "OK" {
+				t.Error("expected successful login")
+			}
+		} else {
+			t.Error("Unexpected response")
 		}
 	case <-timer.C:
 		t.Error("Have not received a response")
 	}
 	timer.Stop()
+	wsAuthExecuted = true
+}
+
+// TestWsPlaceOrder dials websocket, sends order request.
+func TestWsPlaceOrder(t *testing.T) {
+	if !b.Websocket.IsEnabled() && !b.API.AuthenticatedWebsocketSupport || !areTestAPIKeysSet() {
+		t.Skip("API keys not set, skipping")
+	}
+	if !wsAuthExecuted {
+		runAuth(t)
+	}
+	_, err := b.WsNewOrder(&WsNewOrderRequest{
+		CustomID: 1337,
+		Type:     "BUY",
+		Symbol:   "tBTCUSD",
+		Amount:   10,
+		Price:    -10,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+// TestWsCancelOrder dials websocket, sends cancel request.
+func TestWsCancelOrder(t *testing.T) {
+	if !b.Websocket.IsEnabled() && !b.API.AuthenticatedWebsocketSupport || !areTestAPIKeysSet() {
+		t.Skip("API keys not set, skipping")
+	}
+	if !wsAuthExecuted {
+		runAuth(t)
+	}
+	err := b.WsCancelOrder(1234)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+// TestWsCancelOrder dials websocket, sends modify request.
+func TestWsUpdateOrder(t *testing.T) {
+	if !b.Websocket.IsEnabled() && !b.API.AuthenticatedWebsocketSupport || !areTestAPIKeysSet() {
+		t.Skip("API keys not set, skipping")
+	}
+	if !wsAuthExecuted {
+		runAuth(t)
+	}
+	err := b.WsModifyOrder(&WsUpdateOrderRequest{
+		OrderID: 1234,
+		Price:   -111,
+		Amount:  111,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+// TestWsCancelAllOrders dials websocket, sends cancel all request.
+func TestWsCancelAllOrders(t *testing.T) {
+	if !b.Websocket.IsEnabled() && !b.API.AuthenticatedWebsocketSupport || !areTestAPIKeysSet() {
+		t.Skip("API keys not set, skipping")
+	}
+	if !wsAuthExecuted {
+		runAuth(t)
+	}
+	err := b.WsCancelAllOrders()
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+// TestWsCancelAllOrders dials websocket, sends cancel all request.
+func TestWsCancelMultiOrders(t *testing.T) {
+	if !b.Websocket.IsEnabled() && !b.API.AuthenticatedWebsocketSupport || !areTestAPIKeysSet() {
+		t.Skip("API keys not set, skipping")
+	}
+	if !wsAuthExecuted {
+		runAuth(t)
+	}
+	err := b.WsCancelMultiOrders([]int64{1, 2, 3, 4})
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+// TestWsNewOffer dials websocket, sends new offer request.
+func TestWsNewOffer(t *testing.T) {
+	if !b.Websocket.IsEnabled() && !b.API.AuthenticatedWebsocketSupport || !areTestAPIKeysSet() {
+		t.Skip("API keys not set, skipping")
+	}
+	if !wsAuthExecuted {
+		runAuth(t)
+	}
+	err := b.WsNewOffer(&WsNewOfferRequest{
+		Type:   "LIMIT",
```

Comment:
Even though this is fine for testing, we've started using our built in types (order.Limit) in other exchange testing so it's good to keep it consistent